language-plutus-core/src/Language/PlutusCore/Constant/Typed.hs

```diff
@@ -148,6 +149,7 @@ data BuiltinType size
 data TypedBuiltin size a where
     TypedBuiltinSized :: SizeEntry size -> TypedBuiltinSized a -> TypedBuiltin size a
     TypedBuiltinBool  :: TypedBuiltin size Bool
+    TypedBuiltinDyn   :: KnownDynamicBuiltinType dyn => TypedBuiltin size dyn
```

Comment:
There's those existentials!

Reply:
Yes, but they are not in the AST itself, so we still have `Generic`, `NFData`, etc instances for `Term`. We're not even complicating what we already had, because the typed built-ins machinery was already complicated enough (it uses GADTs after all which are strictly more powerful than existentials).